modules/BTBidAdapter.js

```diff
@@ -0,0 +1,237 @@
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { isBoolean, isFn, isStr } from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'blockthrough';
+const GVLID = 815;
+const CURRENCY = config.getConfig('currency.adServerCurrency') || 'USD';
+const ENDPOINT_URL = 'https://pbs.btloader.com/openrtb2/auction';
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid: (bid) => {
+    return (
+      !!bid.params?.siteId &&
+      isStr(bid.params.siteId) &&
+      !!bid.params?.ab &&
+      isBoolean(bid.params.ab)
+    );
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    return validBidRequests.map((validBidRequest) => {
+      const request = getBTRequest(validBidRequest, bidderRequest);
+
+      return {
+        method: 'POST',
+        url: ENDPOINT_URL,
+        data: JSON.stringify(request),
+        bids: validBidRequests,
+      };
+    });
+  },
+
+  interpretResponse: (serverResponse, request) => {
+    if (!serverResponse) {
+      return [];
+    }
+
+    const seatBids = serverResponse?.body.seatbid || [];
+
+    if (!seatBids.length) {
+      return [];
+    }
+    const newBids = seatBids.flatMap((seat) => {
+      return seat.bid.map((bid) => {
+        const bidResponse = getNewBidResponse(bid);
+        bidResponse.btBidderCode = seat.seat;
```

Comment:
Would it make sense to pass in `seat` into `getNewBidResponse` and have `bidResponse.btBidderCode = seat.seat` be done inside the function instead?

Then I think you could do:

```
const newBids = seatBids.flatMap(seat =>
  seat.bid.map(bid => getNewBidResponse(bid,seat))
);
```

Reply:
I'm not super familiar with this, so feel free to leave it as is if you think it makes more sense to keep them separate

Reply:
`bidResponse.btBidderCode = seat.seat;` was done this way, due to Relay implementation specific. so, if we can adjust this and probably rename (not to do `seat.seat`), it would be great! I guess it requires BE change as well.